python/grass/script/core.py

```diff
@@ -1761,16 +1760,40 @@ def create_location(
     if epsg or proj4 or filename or wkt:
         # The names don't really matter here.
         tmp_gisrc, env = create_environment(
-            dbase, "<placeholder>", "<placeholder>", env=env
+            mapset_path.directory, mapset_path.location, mapset_path.mapset, env=env
         )
 
+    # check if location already exists
+    if Path(mapset_path.directory, mapset_path.location).exists():
+        if not overwrite:
+            warning(
+                _("Location <%s> already exists. Operation canceled.")
+                % mapset_path.location,
+                env=env,
+            )
+            return
+        else:
```

Comment:
I missed this during the previous review, this should probably be `error` and not `warning`. Also the `else` is not needed there.

Reply:
Looking at other code there, should it be `fatal(_(msg, env))`? That's what I've put in the latest push.

Reply:
I think you are right, but then you can remove the return.